### PR TITLE
Refactor aria utils and added more test cases

### LINKED ISSUE #576: Reorganize the `ariaHelpers.ts` file and add more test cases
At the moment `ariaHelpers.ts` is a big file that contains some type and some maps of roles/attributes.

The file is getting big and I would like to split it into separate files and try to follow the same approach or [`aria-query`](https://github.com/A11yance/aria-query/tree/master/src/etc/roles) package, although I don't it's necessary to further divide roles into abstract, literal and dpub (especially this last ones, I don't think they are needed at the moment).

As suggested by @bitpshr, we could put everything inside `rules/aria` folder. I just wonder, won't this misled other users/contributors to think that we will have aria rules inside that folder? What do you think?

Suggestions:
- move `ariaPropsMap`inside its own file
- move `roles` inside its own file
- split `roles` in different files, like `aria-query` do. That map will become very big because we will need to support more roles and add more `aria-*` props to each role
- having a file called `utilities.ts, or a folder `utility` with a file per utility. Let me know which one best suits the code style of the repository

At the time being, when I created the lint rules, I tried to implement the most used roles and covered the various combination. Although I think it's best to create more cases in order to cover almost all the roles (at least all the interactive).

The list of the roles can be found here: https://www.w3.org/TR/wai-aria-1.1/#abstract_roles (from there, downward).

May we can create a list of the interactive roles and make sure that are all covered in the lint rules?

### PR DESCRIPTION
**Pull request by ematipico**
Closes #576 

In this PR I decoupled the single file into different files:
- all the types are inside the `types.ts` file
- all the `Map` objects and function are being moved into separate files
- a `index.ts` will export all the former files
- all the functions have a separate file

All the code that uses that single file has been refactored.

Also, more roles have been added. I didn't add ALL the roles. The role that I didn't add are the roles that are abstract or that don't need any special/particular treatment (for now, unless there are some exceptions).

Every time I found a role that needed some required `aria-` attribute, I added a test.
Every time I found a role that had a `baseConcept`, I added a test.

This PR should cover almost all the roles.

### CODE REVIEW

**Comment by ematipico** [comment]
> Thank you for taking care of this! My only minor comment is that I think the imports from `utils/aria/index` and `utils/aria/types` should just import from `utils/aria`.  Other than that, it looks good to me!

Oh yes, that was a problem of the IDE! I will fix it

**Comment by ematipico** [comment]
This is strange. I just rebased my PR and the CI is not passing... but locally it's all fine

**Comment by sebmck** [comment]
I just reran the jobs. It should pass now. There's a test that will sometimes timeout that I'm not sure what the cause is.

File: packages/@romejs/compiler/lint/rules/jsx-a11y/ariaProptypes.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import {Path, TransformExitResult} from "@romejs/compiler";
 import {descriptions} from "@romejs/diagnostics";
 import isEmptyTemplateLiteral from "@romejs/js-ast-utils/isEmptyTemplateLiteral";
+import {ariaPropsMap} from "../../utils/aria";
 import {
 	ARIAProperty,
 	ARIAPropertyDefinition,
-	ariaPropsMap,
-} from "../../utils/aria";
+} from "@romejs/compiler/lint/utils/aria/types";
```
Comment (yassere) [inline packages/@romejs/compiler/lint/rules/jsx-a11y/ariaProptypes.ts:8]: Could you also combine these into a single import statement? Sorry to nitpick!

Comment (ematipico) [reply packages/@romejs/compiler/lint/rules/jsx-a11y/ariaProptypes.ts]: No problem! This is done now!

File: packages/@romejs/compiler/lint/rules/jsx-a11y/ariaProptypes.ts [hunk 2]
```diff
@@ -1,11 +1,11 @@
 import {Path, TransformExitResult} from "@romejs/compiler";
 import {descriptions} from "@romejs/diagnostics";
 import isEmptyTemplateLiteral from "@romejs/js-ast-utils/isEmptyTemplateLiteral";
+import {ariaPropsMap} from "../../utils/aria";
 import {
 	ARIAProperty,
 	ARIAPropertyDefinition,
-	ariaPropsMap,
-} from "../../utils/aria";
+} from "@romejs/compiler/lint/utils/aria";
```
Comment (yassere) [inline packages/@romejs/compiler/lint/rules/jsx-a11y/ariaProptypes.ts:8]: This is still a duplicate import. I think it won't pass CI like this. (It wasn't the macOS timeout problem this time.)

Could you remove the `"../../utils/aria"` import and add `ariaPropsMap` after `ARIAPropertyDefinition` then run `rome check --apply`?

**Pull request merged**